src/librustc_parse/parser/ty.rs

```diff
@@ -304,13 +304,10 @@ impl<'a> Parser<'a> {
         let ast::FnHeader { ext, unsafety, constness, asyncness } = self.parse_fn_front_matter()?;
         let decl = self.parse_fn_decl(|_| false, AllowPlus::No)?;
         let whole_span = lo.to(self.prev_token.span);
-        if let ast::Const::Yes(span) = constness {
-            self.error_fn_ptr_bad_qualifier(whole_span, span, "const");
```

Comment:
The feature gate should be here. There should be a `self.sess.gated_spans.gate(sym::const_fn_pointer, span);` call here and a `gate_all!(const_fn_pointer, "...");` invocation in `src/librustc_ast_passes/feature_gate.rs`.